Modules/PostFilter/ShutterFilter/Filter/ShutterFilter.cs

```diff
@@ -1,11 +1,94 @@
-﻿using VixenModules.OutputFilter.TaggedFilter.Filters;
+﻿using Vixen.Commands;
+using Vixen.Data.Evaluator;
+using Vixen.Data.Value;
+using Vixen.Intent;
+using Vixen.Sys;
+using VixenModules.OutputFilter.TaggedFilter.Filters;
 
 namespace VixenModules.OutputFilter.ShutterFilter.Filter
 {
-	/// <summary>
-	/// Filter that converts color intents into open shutter commands.
-	/// </summary>
-	public class ShutterFilter : TaggedColorFilterBase
+    /// <summary>
+    /// Filter that converts color intents into open shutter commands.
+    /// </summary>
+    public class ShutterFilter : TaggedColorFilterBase
 	{
+		#region Public Properties
+
+		/// <summary>
+		/// Flag which determines if color intents are converted into shutter intents.
+		/// </summary>
+		public bool ConvertColorIntoShutterIntents { get; set; }
+
+		/// <summary>
+		/// Open Shutter index command value.
+		/// </summary>
+		public byte OpenShutterIndexValue { get; set; }
+
+		#endregion
+
+		#region Protected Methods
+
+		/// <summary>
+		/// Refer to base class documentation.
+		/// </summary>
+		public override void Handle(IIntentState<DiscreteValue> obj)
+		{
+			// Handle the discrete color value
+			IntentValue = HandleIntent(obj);
+		}
+
+		/// <summary>
+		/// Refer to base class documentation.
+		/// </summary>
+		public override void Handle(IIntentState<RGBValue> obj)
+		{
+			// Handle the RGB color value
+			IntentValue = HandleIntent(obj);
+		}
+
+		/// <summary>
+		/// Refer to base class documentation.
+		/// </summary>
+		public override void Handle(IIntentState<LightingValue> obj)
+		{
+			// Handle the lighting color value
+			IntentValue = HandleIntent(obj);
+		}
+
+		#endregion
+
+		#region Private Methods
+
+		/// <summary>
+		/// Handles color intents by creating a shutter open command.
+		/// </summary>
+		/// <param name="intent">Intent to handle</param>
+		/// <returns>Handled intent or null if shutter conversion is configured OFF</returns>
+		private IIntentState HandleIntent(IIntentState intent)
+		{
```

Comment:
This is mostly what I expected with the handler, but if the desire is to just drop the intent and build a new one, then why bother passing the intent into a method that to just set it to null? Just let the method be parameterless and either return the new intent or null;

Reply:
Good comment.  I originally was trying to setup something common that would work for this filter, the dimming filter, and the color wheel filter.   The other filters are processing the intent coming in and creating a new one.   Having a common method named HandleIntent did not work out in the end because the other filters have unique code for each incoming color intent type.  I failed to revisit this code and notice the simplification.   Updated as suggested.